### DISCUSSION TITLE
WebGPU: GPUTextureView destruction.

### DISCUSSION THREAD

**Discussion by pplux** [gpuweb/gpuweb#2628] [Q&A]
On the API texture views can be created, but I don't see how texture views can be destroyed. Specially for the webgpu native counterpart.

**Comment by kainino0x** [comment]
They cannot be explicitly destroyed. However, they will become effectively invalid if the underlying GPUTexture object is destroyed.

Only GPUTexture, GPUBuffer, and GPUQuerySet can be explicitly destroyed - and doing so is optional. They hold significant amounts of GPU (VRAM) resources that aren't generally visible as "memory pressure" to a garbage collector, which means GC may not clean them up promptly.

Texture views only reference a GPUTexture object, and do not have a large memory footprint of their own.

**Reply by kainino0x** [reply]
> Specially for the webgpu native counterpart.

WebGPU native is not handled in this repository, but ALL objects in webgpu.h **will** have separate lifetime management in addition to `destroy` methods. The API itself is not resolved yet. https://github.com/webgpu-native/webgpu-headers/issues/9
